### PR TITLE
Enable the newly added OuterPersistent scheduler

### PR DESCRIPTION
**Pull request by liqiangxl**
Enable the newly added OuterPersistent scheduler. It requires outer reduction tvs without inner reduction tvs.
Its priority is higher than the original Persistent scheduler which can process inner, outer, and inner_outer.

### CODE REVIEW

**Comment by liqiangxl** [comment]
!build

File: csrc/kernel_ir.h [hunk 1]
```diff
@@ -1128,7 +1128,7 @@ class GetRNGSeedAndOffsetFromHost : public Expr {
 };
 
 // Expr for driver API cuTensorMapEncodeTiled
-class TORCH_CUDA_CU_API EncodeTensorMapTiled : public Expr {
```
Comment (liqiangxl) [inline csrc/kernel_ir.h:1131]: to pass format check. removed in another PR. https://github.com/NVIDIA/Fuser/pull/921

File: csrc/scheduler/normalization.cpp [hunk 2]
```diff
@@ -46,6 +46,10 @@ void PersistentKernelScheduler::schedule(Fusion* fusion) {
 }
 
 bool PersistentKernelScheduler::canScheduleCompileTime(Fusion* fusion) {
+  // This scheduler is being divided into three separate schedulers and should
```
Comment (liqiangxl) [inline csrc/scheduler/normalization.cpp:49]: Disable this deprecated scheduler.

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build

**Review by naoyam — APPROVED** [review]
LGTM

**Comment by naoyam** [comment]
The codegen diff tests seem to have failed due to a build error:

```
00:01:57 /usr/bin/ld: libnvfuser_codegen.so: undefined reference to `cuTensorMapEncodeTiled'
```

This seems to be related to the recent TMA PR by Xiang, but I don't know why the error only shows up now. Any idea? @zasdfgbnm, @xwang233

**Comment by zasdfgbnm** [comment]
@xwang233 Are we using ccache in our build, and would it help to clean all the cache? I used to use ccache and it used to cause weird error like this and that's why I stopped using it anymore. But this experience was 10+ years ago.

**Comment by xwang233** [comment]
@zasdfgbnm , hey, I guess the cause of those build issues in codegen diff jobs is that, the base image we used to build your PR was from last night where the TMA PR was not merged to main yet. Any new PR today, after the TMA PR got merged, would be built on top of the "old base image", where the old CMake cache may cause weird conflicts with the new PR.

I suggest that we wait until tomorrow when the new base image, built from main, contains the TMA PR. Those cuTensor related dependency issue should be resolved by then. I'll take a look if CI jobs still have this issue tomorrow and I have some rough ideas on where could go wrong.

Ccache is used in our test_container build but **not** in codegen_diff jobs or manylinux_wheel builds. 🙂

**Comment by zasdfgbnm** [comment]
> @zasdfgbnm , hey, I guess the cause of those build issues in codegen diff jobs is that, the base image we used to build your PR was from last night where the TMA PR was not merged to main yet. Any new PR today, after the TMA PR got merged, would be built on top of the "old base image", where the old CMake cache may cause weird conflicts with the new PR.
> 
> I suggest that we wait until tomorrow when the new base image, built from main, contains the TMA PR. Those cuTensor related dependency issue should be resolved by then. I'll take a look if CI jobs still have this issue tomorrow and I have some rough idea on where could go wrong.
> 
> Ccache is used in our test_container build but **not** in codegen_diff jobs or manylinux_wheel builds. 🙂

Sounds like a valid theory. Thanks!

**Comment by xwang233** [comment]
I think the missing dependency issue in compilation is due to too many concurrency, e.g., 255 jobs there, in the build. The concurrency algorithm (MAX_JOBS) might be fine for local build, but doesn't work that well in servers with large memory and multiple containers running simultaneously. I've reduced the building concurrency in CI codegen diff jobs. Hope this can fix the build issue.

**Pull request merged**